lib/ansible/modules/system/user.py

```diff
@@ -946,13 +954,22 @@ def modify_user(self):
                     new_groups = groups | set(current_groups)
                 cmd.append(','.join(new_groups))
 
-        if self.expires:
-            current_expires = time.gmtime(int(self.user_password()[1]))
+        if self.expires is not None:
+
+            current_expires = self.user_password()[1]
```

Comment:
This value is a string in BSD, which means the conditional on line 962 will evaluate to True when it should not.

`0.0s modify_user: type(current_expires)=<type 'str'>`

Reply:
After looking at this a bit, more, it's a string because `HAS_SPWD` is False, which is causing it to parse the shadow file directly and return a string. If I set `HAS_SPWD` to `False` and run against a Linux distro, it fails with `ValueError: invalid literal for int() with base 10: ''`.